### PR TITLE
fix missing request attributes when using auth-method CLIENT-CERT

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java [hunk 1]
```diff
@@ -70,15 +71,20 @@ public Authentication validateRequest(Request req, Response res, Callback callba
             Response.writeError(req, res, callback, HttpServletResponse.SC_FORBIDDEN);
             return Authentication.SEND_FAILURE;
         }
-        
+
+        if (sslSessionData.cipherSuite() != null)
+        {
+            req.setAttribute("jakarta.servlet.request.cipher_suite", sslSessionData.cipherSuite());
+        }
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java:78]: This is wrong, as the `cipherSuite` has already been set as a request attribute by the `SecureRequestCustomizer`.
Why doing this here?

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java [hunk 2]
```diff
@@ -70,15 +71,20 @@ public Authentication validateRequest(Request req, Response res, Callback callba
...
+        req.setAttribute("jakarta.servlet.request.key_size", sslSessionData.keySize());
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java:79]: Ditto.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java [hunk 3]
```diff
@@ -70,15 +71,20 @@ public Authentication validateRequest(Request req, Response res, Callback callba
...
         X509Certificate[] certs = sslSessionData.peerCertificates();
-        
+
         try
         {
             // Need certificates.
             if (certs != null && certs.length > 0)
             {
-
+                req.setAttribute("jakarta.servlet.request.X509Certificate", certs);
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java:87]: Ditto.

File: jetty-ee9/jetty-ee9-security/src/main/java/org/eclipse/jetty/ee9/security/authentication/SslClientCertAuthenticator.java [hunk 4]
```diff
@@ -63,14 +64,27 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
...
+        req.setAttribute("jakarta.servlet.request.key_size", sslSessionData.keySize());
+        X509Certificate[] certs = sslSessionData.peerCertificates();
 
         try
         {
             // Need certificates.
             if (certs != null && certs.length > 0)
             {
-
+                req.setAttribute("jakarta.servlet.request.X509Certificate", certs);
```
Comment (sbordet) [inline jetty-ee9/jetty-ee9-security/src/main/java/org/eclipse/jetty/ee9/security/authentication/SslClientCertAuthenticator.java:87]: Same as for EE10. This class should be unchanged, only `SecureRequestCustomizer` should be changed.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-ee9/jetty-ee9-security/src/main/java/org/eclipse/jetty/ee9/security/authentication/SslClientCertAuthenticator.java (hunk 4 above)
Comment (joakime) [reply jetty-ee9/jetty-ee9-security/src/main/java/org/eclipse/jetty/ee9/security/authentication/SslClientCertAuthenticator.java:87]: Do we have a ee10 specific `SecureRequestCustomizer` ?

Comment (sbordet) [reply jetty-ee9/jetty-ee9-security/src/main/java/org/eclipse/jetty/ee9/security/authentication/SslClientCertAuthenticator.java:87]: We do not, but we should at this point.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java (hunk 1 above)
Comment (olamy) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java:78]: I'm not quite sure to understand. https://github.com/eclipse/jetty.project/blob/jetty-12.0.x/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java is in jetty-core so this class doesn’t have to set any servlet attribute (as could be javax or jakarta).

Comment (sbordet) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/security/authentication/SslClientCertAuthenticator.java:78]: I think we need to write EE secure customizers.

**Comment by olamy** [comment]
as we need to write some EEs secure customizers. 
Should we have them in `jetty-eeX-servlet` and some `eeX-ssl.mod`?

**Comment by sbordet** [comment]
@olamy I'll prepare a PR about EE secure customizers.

**Comment by sbordet** [comment]
@olamy please look at #9533 instead of this PR.
Does that PR pass TCK?

**Comment by olamy** [comment]
replaced by https://github.com/eclipse/jetty.project/pull/9533

**Pull request closed without merging**